apis/hive/v1/generics.go

```diff
@@ -0,0 +1,9 @@
+package v1
+
+type Condition interface {
+	ConditionType() ConditionType
+}
+
+type ConditionType interface {
+	AsString() string
```

Comment:
Out of curiosity, what happens if you do one of these (mutually exclusive) alternatives?
- Use the existng [fmt.Stringer interface](https://pkg.go.dev/fmt#Stringer)
- Nix the interface and just use `string()` inline in `FindCondition`

?

Reply:
I forgot about the `fmt.Stringer` interface... I think I'll do something like this 

```go
type ConditionType interface {
	fmt.Stringer
}
```

so that `ConditionType` can still be used as the generic constraint on `FindCondition`.  Alternatively, `FindCondition` could take `Stringer`s instead of `ConditionType`s

Reply:
I think I prefer the former. The latter would mean you would have to change every condition type to a Stringer, which isn't _awful_, but if it's gonna be an interface type anyway, I like the explicitness of `ConditionType`.

That said, what about my second bullet? Then all the condition types can just be `string` and we don't need an interface for it.